app/forms/reservation_form.py

```diff
@@ -34,7 +34,7 @@ def validate_occasion(form, field):
     Validate that occasion_id points to a valid occasions entry
     """
     occasion = Occasion.query.get(field.data)
-    if occasion is None:
+    if occasion is not None:
```

Comment:
Why would this be not None if it is raising a validation error for occasion being missing?